.travis.yml

```diff
@@ -10,13 +10,16 @@ before_install:
   - export PATH=${HOME}:${PATH}
   - export OPAMYES=1
   - export OPAMJOBS=2
-  - opam init --compiler=${OCAML_VERSION} --disable-sandboxing
+  - opam init --bare --disable-sandboxing
+  - opam repo add --dont-select beta https://github.com/ocaml/ocaml-beta-repository.git
+  - opam switch --dry-run ${OCAML_VERSION} 2>/dev/null || opam switch create ${OCAML_VERSION} --repositories=default,beta
+  - opam switch ${OCAML_VERSION}
   - eval `opam config env`
   - export OPAMVERBOSE=1
 install:
-  - opam update --upgrade
-  - opam pin add --no-action ocamlformat .
-  - opam install --deps-only ocamlformat
+  - opam config exec --switch=${OCAML_VERSION} -- opam update --upgrade
```

Comment:
I think this is not correct. The `opam update` command will not look at the environment variables and will use the switch selected above.

On my machine, `opam config exec --switch=4.08.0+beta3 -- opam switch show` gives `4.07.1`, which is my current switch.

You should either pass the `--switch` option directly to `opam update` or move the `opam switch ${OCAML_VERSION}` command into the `install` section.

Reply:
You are still using `opam config exec`, I think it make no sense.